docs/src/components/quick-nav/QuickNav.tsx

```diff
@@ -250,6 +250,11 @@ function onMounted(ref: React.RefObject<HTMLDivElement | null>) {
   };
 }
 
+/** Get window.scrollY accounting for a potential scroll lock offset top style */
+function getScrollY() {
+  return window.scrollY - parseFloat(document.documentElement.style.top || '0');
+}
```

Comment:
This scroll locking technique is looking to cause a bunch of interop issues with positioning math, such as in custom extensions or code on users' websites. I guess we'll need to see how large the impact is, but it may not be the bulletproof solution we're looking for compared to a simple `overflow: hidden` (at least on non-iOS browsers...)

Reply:
Let's discuss it today with https://github.com/mui/base-ui/pull/1098